component/plugincontroller/component.go

```diff
@@ -0,0 +1,34 @@
+package plugincontroller
+
+import (
+	"github.com/docker/docker/component"
+	"github.com/docker/docker/plugin"
+	"golang.org/x/net/context"
+)
+
+const name = "plugincontroller"
+
+// Set sets the plugin controller component
+func Set(s *plugin.Manager) (cancel func(), err error) {
+	return component.Register(name, s)
+}
```

Comment:
Isn't the problem right now that the daemon is responsible for way too may things. A small class which handles coordination between components would be fine, but a large class that handles coordination plus lots of "core application logic" is a nightmare.

All the application logic would move to components which leaves us with a very small coordinator.